examples/README.md

```diff
@@ -0,0 +1,134 @@
+# nRF 802.15.4 radio driver - examples
+
+Example bare-metal applications for nRF IEEE 802.15.4 radio driver are provided to familiarize you
+with the radio driver API and reduce amount of work required to create your own IEEE 802.15.4
+compliant application.
+
+## Required hardware
+
+Provided examples are compatible with the nRF52840 DK.
+
+## Toolchain
+
+Tools required to build the examples and program the board are as follows.
+
+**CMake** - version 3.13.1 or above - is required to generate the build scripts for a build tool of
+choice. This manual uses `make`.
+
+**ARMGCC** - version 9-2019-q4-major - is required to build the examples. Provided CMake files assume
+the ARM Toolchain is included in the path.
+
+**nRF5x Command Line Tools** - version 9.4.0 - are required as they provide `nrfjprog` utility used
+to program the target device.
+
+## Creating an out-of-tree application
+
+The top-level CMake description makes no assumptions about the project directory hierarchy using the nRF 802.15.4 radio driver, but expects the CMake description pulling the driver to meet certain requirements.
+
+The CMake description that pulls the radio driver must set the following parameters prior to the directory inclusion:
+* `TARGET_SOC` variable - specifies the SoC that the radio driver is meant to be built for.
+* `TARGET_ABI` variable - specifies the float ABI that the radio driver is meant to use.
+* `nrf_802154_platform` target - the target must provide all dependencies necessary to build the radio driver:
+   - nrfx library
+   - mpsl library
+   - compiler ABI flags
+   - must contain platform abstraction source files
+   - must set the NRF_802154_INTERNAL_SWI_IRQ_HANDLING compile definition
+
+   The target must be a static library.
+
+Example integration with an application:
+```cmake
+cmake_minimum_required(VERSION 3.13.1)
+project(application)
+
+#
+# Add nrfx drivers. The target can be named anything, as long as the headers and library
+# is visible to the radio driver top-level CMake.
+#
+add_library(nrfx-custom STATIC "")
+
+# Add any nrfx files, header paths, compiler options, defines, ABI settings etc. that are
+# necessary to compile and link the application.
+target_xyz(...)
+
+#
+# Include a CMake file which sets NRF_802154_DRIVER_PLATFORM_BAREMETAL
```

Comment:
If that's an option, I'd be in favour of renaming this CMake variable to `NRF_802154_DRIVER_SOURCES_PLATFORM_BAREMETAL` so that:
* the name clearly implies a list of source files, not e.g. a boolean that indicates whether platform type is baremetal
* the name matches its SL counterpart

Reply:
Variable was renamed as suggested.